docs/design/2022-08-02-stats-lru-cache.md

```diff
@@ -0,0 +1,69 @@
+# Proposal: Stats LRU Cache
+
+- Author(s): [Yisaer](https://github.com/yisaer)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/34052
+
+## Abstract
+
+This proposes a design of how to maintain stats cache by lru according to memory usage
+
+## Background
+
+Previously, tidb maintained the all the indices' stats and some columns' stats which is needed during query. 
+As the maintained stats grows, the total memory usage of the stats will increase and makes tidb server OOM.
+Thus we use lru to maintain stats cache in order to keep memory safe.
+
+### Goal
+
+- Use LRU to maintain the stats cache in memory
+- Keep the total memory usage of stats cache under the quota
+
+### Non-Goals
+
+- Considering the stats cache are in memory, we don't provide changing stats cache into LRU without restarting
+
+## Proposal
+
+### Stats Cache Interface
+
+We will provide a Stats Cache Interface which is implemented by LRU Cache and Map Cache. 
+If the tidb server didn't enable Stats LRU Cache, it will use Map Cache by default.
```

Comment:
What about introducing the variables to turn on LRU and to set the capacity?

Reply:
updated.